spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/converter/JacksonPubSubMessageConverter.java

```diff
@@ -53,18 +46,32 @@ public JacksonPubSubMessageConverter(ObjectMapper objectMapper) {
 	}
 
 	@Override
-	public byte[] toPayload(Object object) throws IOException {
+	public PubsubMessage toMessage(Object payload, Map<String, String> headers) {
 		try {
-			return this.objectMapper.writeValueAsBytes(object);
+			PubsubMessage.Builder pubsubMessageBuilder = PubsubMessage.newBuilder()
+					.setData(ByteString.copyFrom(this.objectMapper.writeValueAsBytes(payload)));
+
+			if (headers != null) {
+				pubsubMessageBuilder.putAllAttributes(headers);
+			}
+
+			return pubsubMessageBuilder.build();
 		}
 		catch (JsonProcessingException e) {
-			throw new IOException("Jackson serialization of object failed: " + object, e);
+			throw new RuntimeException("JSON serialization of an object of type " +
```

Comment:
I suggest to throw here an explicit `MessageConversionException` instead. It is indeed the conversion issue. Isn't it?

Reply:
`MessageConversionException` belongs to `spring-messaging`. Do we want the template to depend on it?

Reply:
Well, that's fine. The `spring-messaging` is a generic dependency - there is nothing about separate Spring Integration project.

Reply:
The description of `MessageConversionException` is `An exception raised by {@link MessageConverter} implementations.` Since we do not implement `MessageConverter`, I think it might be misleading.

Reply:
Yeah... Agreed - unfortunate to have such a JavaDocs there. Although it doesn't break a purpose and functionality of the exception. So, what do you suggest: introduce our own exception on the matter?

I just don't like such a generic `RuntimeException` just for the goal to hide checked exception. 😢

Reply:
Our own will do. :)